services/resource-usage-tracker/tests/unit/with_dbs/test_collect_container_resource_usage_task.py

```diff
@@ -0,0 +1,57 @@
+import json
+from pathlib import Path
+from unittest import mock
+
+import pytest
+import sqlalchemy as sa
+from fastapi import FastAPI
+from pytest_mock import MockerFixture
+from simcore_postgres_database.models.resource_tracker import resource_tracker_container
+from simcore_service_resource_usage_tracker.resource_tracker_core import (
+    collect_container_resource_usage_task,
+)
+
+pytest_simcore_core_services_selection = [
+    "postgres",
+]
+pytest_simcore_ops_services_selection = [
+    "adminer",
+]
+
+
+@pytest.fixture
+def mocked_prometheus_client_custom_query(
+    mocker: MockerFixture, project_slug_dir: Path
+) -> mock.MagicMock:
+    with open(
+        project_slug_dir / "tests" / "unit" / "with_dbs" / "data" / "data.json"
+    ) as file:
+        data = json.load(file)
+
+    mocked_get_prometheus_api_client = mocker.patch(
+        "simcore_service_resource_usage_tracker.resource_tracker_core._prometheus_client_custom_query",
+        autospec=True,
+        return_value=data,
+    )
+    return mocked_get_prometheus_api_client
+
+
+async def test_collect_container_resource_usage_task(
+    mocked_prometheus: mock.Mock,
+    mocked_prometheus_client_custom_query: mock.MagicMock,
+    initialized_app: FastAPI,
+    postgres_db: sa.engine.Engine,
+):
+    await collect_container_resource_usage_task(initialized_app)
+
+    expected_query = "sum without (cpu) (container_cpu_usage_seconds_total{image=~'registry.osparc.io/simcore/services/dynamic/jupyter-smash:.*'})[30m:1m]"
+    mocked_prometheus_client_custom_query.assert_called_once_with(
```

Comment:
I will comment about this here.

Say the application goes down, and you lack the last 4 hours of scraping. Having a fixed interval is not a good idea. I would look at the last changed date time in the debase and use that one to figure out from where to scrap. I will do that on every scraping action. This way you are free to run the scarping at whatever interval you please and then you are sure it will fetch all interesting data.

Reply:
Cool, you are right! It was added to the Next steps which you can see in the top comment.